Meta: use latest ubuntu image for ci

sandr01d opened the pull request:
## Check list

- [X] I have performed a self-review of my code
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

## Description

GitHub has deprecated the `ubuntu-20.04` runnner as outlined [here](https://github.com/actions/runner-images/issues/11101). This PR updates the CI to always include the latest available Ubuntu image (currently `ubuntu-24.04`) and removes `ubuntu-20.04`.

We should probably merge this PR before anything else, to prevent the CI from failing when merging new PRs.

## Type of change

- [ ] Bug fix
- [ ] New feature
- [ ] Refactor
- [ ] Breaking change
- [ ] Documentation change
- [X] CI

## Test environment

- Shell
    - [ ] bash
    - [ ] zsh
    - [ ] fish
- OS
    - [ ] Linux
    - [ ] Mac OS X
    - [ ] Windows
    - [ ] Others:

wfxr reviewed (approved):
LGTM

carlfriedrich commented:
Thanks for the update @sandr01d. While we're at it: is there a reason why we're using two different Ubuntu versions for the CI run? For MacOS we're using only the latest version, so why not drop the static ubuntu-22.04 version as well and use only ubuntu-latest?

sandr01d commented:
> Thanks for the update @sandr01d. While we're at it: is there a reason why we're using two different Ubuntu versions for the CI run? For MacOS we're using only the latest version, so why not drop the static ubuntu-22.04 version as well and use only ubuntu-latest?

True, there is nothing in the CI that should differentiate between varying version of Ubuntu. I've moved ahead and removed `ubuntu-22.04`, keeping only `ubuntu-latest`.
